js/utility/regex-tools.js

```diff
@@ -0,0 +1,23 @@
+namespacer('bcpl.utility');
+
+bcpl.utility.regexTools = (() => {
+	/**
+	 * Filters a string down to only characters that match the regex.
+	 *
+	 * @param {string} stringToFilter
+	 * @param {RegExp} filterRegex
+	 */
+	const filter = (stringToFilter, filterRegex) => {
```

Comment:
i think we could name this better (I think), not sure what it would be.

Reply:
renamed!